packages/react-native/ReactCommon/react/performance/timeline/PerformanceObserverRegistry.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <set>
+#include "PerformanceObserver.h"
+
+namespace facebook::react {
+
+/**
+ * PerformanceObserverRegistry acts as a container for known performance
+ * observer instances.
+ *
+ * You can queue performance entries through this registry, which then delegates
+ * the entry to all registered observers.
+ */
+class PerformanceObserverRegistry {
+ public:
+  using Ptr = std::unique_ptr<PerformanceObserverRegistry>;
+  using WeakPtr = std::weak_ptr<PerformanceObserverRegistry>;
+
+  PerformanceObserverRegistry() = default;
+
+  /**
+   * Creates Performance Observer instance and registers it in the registry.
+   *
+   * Observer is automatically removed from the registry when it goes out of scope.
```

Comment:
I think we should separate the observer lifecycle from its presence in the registry. We should create the observer independently as a shared pointer (so we can use it as native state), and when we start observing with that observer we should add it to the registry (and the registry should retain the observer, until we disconnect the observer from the registry). This will be easier to reason about and would align with the spec.